packages/flutter_tools/test/general.shard/ios/devices_test.dart

```diff
@@ -64,34 +70,187 @@ void main() {
+        platform: macPlatform,
+        iosDeploy: iosDeploy,
+        name: 'iPhone 1',
+        sdkVersion: '13.3',
+        cpuArchitecture: DarwinArch.arm64
+      );
+    });
+
+    testWithoutContext('parses major version', () {
+      expect(IOSDevice(
+        'device-123',
+        artifacts: mockArtifacts,
+        fileSystem: mockFileSystem,
+        platform: macPlatform,
+        iosDeploy: iosDeploy,
+        name: 'iPhone 1',
+        cpuArchitecture: DarwinArch.arm64,
+        sdkVersion: '1.0.0'
+      ).majorSdkVersion, 1);
+      expect(IOSDevice(
+        'device-123',
+        artifacts: mockArtifacts,
+        fileSystem: mockFileSystem,
+        platform: macPlatform,
+        iosDeploy: iosDeploy,
+        name: 'iPhone 1',
+        cpuArchitecture: DarwinArch.arm64,
+        sdkVersion: '13.1.1'
+      ).majorSdkVersion, 13);
+      expect(IOSDevice(
+        'device-123',
+        artifacts: mockArtifacts,
+        fileSystem: mockFileSystem,
+        platform: macPlatform,
+        iosDeploy: iosDeploy,
+        name: 'iPhone 1',
+        cpuArchitecture: DarwinArch.arm64,
+        sdkVersion: '10'
+      ).majorSdkVersion, 10);
+      expect(IOSDevice(
+        'device-123',
+        artifacts: mockArtifacts,
+        fileSystem: mockFileSystem,
+        platform: macPlatform,
+        iosDeploy: iosDeploy,
+        name: 'iPhone 1',
+        cpuArchitecture: DarwinArch.arm64,
+        sdkVersion: '0'
+      ).majorSdkVersion, 0);
+      expect(IOSDevice(
+        'device-123',
+        artifacts: mockArtifacts,
+        fileSystem: mockFileSystem,
+        platform: macPlatform,
+        iosDeploy: iosDeploy,
+        name: 'iPhone 1',
+        cpuArchitecture: DarwinArch.arm64,
+        sdkVersion: 'bogus'
+      ).majorSdkVersion, 0);
     });
 
     for (final Platform platform in unsupportedPlatforms) {
-      testUsingContext('throws UnsupportedError exception if instantiated on ${platform.operatingSystem}', () {
+      testWithoutContext('throws UnsupportedError exception if instantiated on ${platform.operatingSystem}', () {
         expect(
-          () { IOSDevice('device-123', name: 'iPhone 1', sdkVersion: '13.3', cpuArchitecture: DarwinArch.arm64); },
+          () {
+            IOSDevice(
+              'device-123',
+              artifacts: mockArtifacts,
+              fileSystem: mockFileSystem,
+              platform: platform,
+              iosDeploy: iosDeploy,
+              name: 'iPhone 1',
+              sdkVersion: '13.3',
+              cpuArchitecture: DarwinArch.arm64,
+            );
+          },
           throwsAssertionError,
         );
-      }, overrides: <Type, Generator>{
-        Platform: () => platform,
       });
     }
 
+    group('ios-deploy wrappers', () {
+      const String appId = '789';
+      IOSDevice device;
+      IOSDeploy iosDeploy;
+      FullMockProcessManager mockProcessManager;
+
+      setUp(() {
+        mockProcessManager = FullMockProcessManager();
```

Comment:
consider using FakeProcessManager.list/.any instead of a Mock

Reply:
@jonahwilliams The reason I created a mock is because I wanted to have `processManager.run()` throw an exception, and `FakeProcessManager` already has an implementation of `.run()`. Is there a way to get a `FakeProcessManager` to throw?

Reply:
I think you can throw an exception in `onRun`, but I might be mistaken. if not we should add that feature